### PR TITLE
Update scala queries

### PR DESCRIPTION
**Pull request by stevanmilic**
Add scala folds

Update scala highlights

* add highlights for imports

* add highlights for type constructors

* add highlights for constants

* add highlights for interpolation string

* add highlights for comments through injection

### CODE REVIEW

**Comment by vigoux** [comment]
I see that scala does not yet have a maintainer. If you'd like to become one, that would greatly help us, and for you that basically means that when the scala parser has an update with breaking change, we will ping you for help to fix the queries.

**Comment by stevanmilic** [comment]
Sure @vigoux, I'll be glad.

File: queries/scala/highlights.scm [hunk 1]
```diff
@@ -7,6 +7,15 @@
 ((identifier) @variable.builtin
  (#match? @variable.builtin "^this$"))
 
+; Assume other uppercase names constants.
+; NOTE: In order to distinguish constants we highlight
+; all the identifiers that are uppercased. But this solution
+; is not suitable for all occurences e.g. it will highlight
+; an uppercased method as a constant if used with no params.
+; Introducing highlighting for those specifi cases, is probably
```
Comment (theHamsta) [inline queries/scala/highlights.scm:15]: Typo

File: queries/scala/highlights.scm [hunk 2]
```diff
@@ -7,6 +7,15 @@
...
+; best way to resolve the issue.
+((identifier) @constant (#match? @constant "^[A-Z]"))
```
Comment (theHamsta) [inline queries/scala/highlights.scm:17]: So in Scala you have constants identifiers like `Pi`, `MaxVertices`? Or rather `PI` , `MAX_VERTICES`

File: queries/scala/highlights.scm [hunk 3]
```diff
@@ -77,14 +106,17 @@
...
 
 [
 (symbol_literal)
 (string)
 (character_literal)
+(interpolated_string_expression)
 ] @string
 
+(interpolation "$" @punctuation.special)
+(interpolation (identifier) @variable)
```
Comment (theHamsta) [inline queries/scala/highlights.scm:118]: Can't this be treated like interpolations in Python or Bash? The variables should get `@variable` automatically when the string highlight is removed and the interpolation query is placed on the top of the file. Maybe this does not work for Scala

File: queries/scala/highlights.scm (hunk 2 above)
Comment (stevanmilic) [reply queries/scala/highlights.scm:17]: `Pi` yes, check the [offical docs ](https://docs.scala-lang.org/style/naming-conventions.html#constants-values-and-variables).

File: queries/scala/highlights.scm (hunk 3 above)
Comment (stevanmilic) [reply queries/scala/highlights.scm:118]: Yes, I was wondering what's the best approach for interpolation. I'll try what you proposed.

Comment (stevanmilic) [reply queries/scala/highlights.scm]: It's looking great with `@none` now, thanks!

**Review by theHamsta — APPROVED** [review]
Looks nice! Thanks for your contribution!

**Pull request merged**